### ISSUE TITLE
DataCollection and FileAccess are not serializable

### ISSUE THREAD

**Issue by egorsobolev** [European-XFEL/EXtra-data#22]
The reason is the underlying h5py.File attribute in FileAccess which is not serializable. It would be good to add `__get_state__` method to FileAccess like

```python
def __get_state__(self):
    state = self.__dict__.copy()
    state['_file'] = None
    return state
```

**Comment by takluyver** [comment]
Do you mean with pickle? What are you trying to do at a higher level?

**Comment by egorsobolev** [comment]
Yes, with pickle. I am trying to transfer the DataCollection instance from subprocess to the main process. It is a hack to kill the process which hangs up due to io operations on temporarily unavailable data.

**Comment by takluyver** [comment]
That sounds... exciting. In any case, yes, feel free to make a PR to add a `__getstate__` method.

**Comment by takluyver** [comment]
This got fixed in #25.
